### DISCUSSION TITLE
How to remove 'runtimeType' key from toJson?

### DISCUSSION THREAD

**Discussion by Saltitans** [rrousselGit/freezed#769] [Q&A]
I have a situation where I'm serializing a union object so I can make an HTTP request. 

The issue is that when doing 'toJson()', the 'runtimeType' key is added making the API throw an error because there's unrecognized content on the request.

Is there a way to ignore the 'runtimeType' key the same way we do with 'JsonKey(ignore: true)' for the other attributes?

**Comment by rrousselGit** [comment]
You can override toJson to manually remove the runtimeType key

Although you'll likely need a custom toJson too if your API doesn't return this key, because then Freezed won't know how to decode the union.

**Reply by Saltitans** [reply]
Thank you for the answer!

I followed your suggestion and override the toJson method in the union class. Since I don't need a fromJson, there's no need to worry about its implementation.

Despite trying to override toJson, the generated methods are still being called instead of my override.

Here's an example of my approach:

```
@freezed
class ClassA with _$ClassA {
  const factory ClassA.type1(
    String a,
  ) = _ClassAType1;

  const factory ClassA.type2({
    required String b,
    required String c,
  }) = _ClassAType2;

  factory ClassA.fromJson(Map<String, dynamic> json) => _$ClassAFromJson(json);

  @override
  Map<String, dynamic> toJson() => when(
        type1: (a) => <String, dynamic>{
          'a': a
        },
        type2: (b, c) => <String, dynamic>{
          'b': b,
          'c': c,
        },
      );
}
```

Could you point me in the right direction, please?

**Reply by rrousselGit** [reply]
See the Readme: when defining methods/getters, you need to also define an empty private constructor

**Reply by Saltitans** [reply]
My bad, I forgot to add it to the example but I have the empty private constructor defined and the generated methods are still being called instead of my override.

Edit: I've created a repo [here](https://github.com/Saltitans/test_app) with a small example app where I show this issue if you want to take a look!
